dataflow/src/org/checkerframework/dataflow/analysis/FlowExpressions.java

```diff
@@ -407,7 +407,8 @@ private static Receiver internalRepOfMemberSelect(
             case ENUM:
             case INTERFACE: // o instanceof MyClass.InnerInterface
             case ANNOTATION_TYPE:
-                return new ClassName(expressionType);
+                TypeMirror selectType = TreeUtils.typeOf(memberSelectTree);
+                return new ClassName(selectType);
             case ENUM_CONSTANT:
             case FIELD:
                 Receiver r = internalReprOf(provider, memberSelectTree.getExpression());
```

Comment:
The line below this one (on which I can't comment) uses `ElementUtils.getType(ele)`.
This seems wrong or at least not as precise as possible. I would use `TreeUtils.typeOf(memberSelectTree)`.
The difference is that the type of the field from the Element is the type from the declaration of the field, which could e.g. be a type parameter. The actual type from the `memberSelectTree` should be the fully-substituted type.